pkg/feeder/types.go

```diff
@@ -142,18 +143,40 @@ type TransactionExecution struct {
 	Events []Event `json:"events"`
 	// The resources needed by the transaction.
 	ExecutionResources ExecutionResources `json:"execution_resources"`
+	ActualFee          string             `json:"actual_fee"`
 }
 
 // StarknetBlock Represents a response StarkNet block.
 type StarknetBlock struct {
-	BlockHash           string               `json:"block_hash"`
-	ParentBlockHash     string               `json:"parent_block_hash"`
-	BlockNumber         types.BlockNumber    `json:"block_number"`
-	StateRoot           string               `json:"state_root"`
-	Status              rpc.BlockStatus      `json:"status"`
-	Transactions        TxnSpecificInfo      `json:"transactions"`
-	Timestamp           int64                `json:"timestamp"`
-	TransactionReceipts TransactionExecution `json:"transaction_receipts"`
+	BlockHash           string                 `json:"block_hash"`
+	ParentBlockHash     string                 `json:"parent_block_hash"`
+	BlockNumber         types.BlockNumber      `json:"block_number"`
+	GasPrice            string                 `json:"gas_price"`
+	SequencerAddress    string                 `json:"sequencer_address"`
+	StateRoot           string                 `json:"state_root"`
+	Status              rpc.BlockStatus        `json:"status"`
+	Transactions        []TxnSpecificInfo      `json:"transactions"`
+	Timestamp           int64                  `json:"timestamp"`
+	TransactionReceipts []TransactionExecution `json:"transaction_receipts"`
+}
+
+//struct to store Storage info
+type StorageInfo struct {
+	Storage string `json:"storage"`
+}
+
+//ABI input struct
+type Input struct {
```

Comment:
If this struct is used to map all the items of an ABI I think that is not all the possible fields. The ABI is an array of items where each item can be one of these types:
- Function
- Event
- Struct
- L1Handler (the same structure of a Function)
- Constructor (the same structure of a Function)

> See this [ABI doc](https://docs.starknet.io/docs/Contracts/contract-abi) example and [this](https://alpha-mainnet.starknet.io/feeder_gateway/get_code?contractAddress=0x20cfa74ee3564b4cd5435cdace0f9c4d43b939620e4a0bb5076105df0a626c6) mainnet example.

So types like Event and Struct do not match there, and at the same time, we don't want one struct to handle all the possible types because then we will manage many `if` conditions over the `Type` field, and unnecessary fields for certain types. A solution is, maybe, a struct like this:
```
type ABI struct {
    Functions []Function
    Events.     []Event
    Structs.    []Structs
    L1Handlers  []L1Handler
    Constructor Function
}
```

And implement the for that `ABI` struct the `json.Unmarshaler` interface.

Reply:
This is good guidance - I will implement it.